### PR TITLE
[L1T] Update track z0 in the fastHistoEmulation algorithm of VertexFinder

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This PR makes some changes to how the vertexing algorithm treats the track z0 parameter. It allows for the use of the full dynamic range of the track parameters. Additionally, this PR updates some of the ancillary code used by the GTT developers.

Porting local https://github.com/cms-l1t-offline/cmssw/pull/1037

#### PR validation

This PR was validated by passing the resulting test vectors through the Xilinx HLS firmware generation stage. We also performed many of the calculations by hand in order to validate that the code was performing the correct operations. Finally, we ran the scram code-checks and code-format commands to make sure the code complied with the CMS code rules.

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

File: L1Trigger/DemonstratorTools/test/gtt/createFirmwareInputFiles_cfg.py [hunk 1]
```diff
@@ -44,29 +44,31 @@
 
 process = cms.Process("GTTFileWriter")
 
-process.load('Configuration.Geometry.GeometryExtended2026D49Reco_cff')
-process.load('Configuration.Geometry.GeometryExtended2026D49_cff')
+process.load('Configuration.Geometry.GeometryExtended2026D77Reco_cff')
```
Comment (srimanob) [inline L1Trigger/DemonstratorTools/test/gtt/createFirmwareInputFiles_cfg.py:47]: Please move to D86 or D88 or later.

**Comment by cecilecaillol** [comment]
please test

File: L1Trigger/DemonstratorTools/test/gtt/createFirmwareInputFiles_cfg.py (hunk 1 above)
Comment (cecilecaillol) [reply L1Trigger/DemonstratorTools/test/gtt/createFirmwareInputFiles_cfg.py]: Done

**Comment by cecilecaillol** [comment]
+l1

**Comment by srimanob** [comment]
+Upgrade

Is this needed for 12_5 production?

**Comment by perrotta** [comment]
+1

**Comment by epalencia** [comment]
> +Upgrade
> 
> Is this needed for 12_5 production?

Hi @srimanob , no, this one is not needed.

**Pull request merged**